File: server/src/controller/health.controller.ts
import { Controller, Get } from "@midwayjs/core";
import { ApiTags, ApiOperation } from "@midwayjs/swagger";
import { Public } from "../decorator/public.decorator";
import { ResponseUtil } from "../common/types/response";

@ApiTags("health")
@Controller("/api")
export class HealthController {
  @Public()
  @Get("/health")
  @ApiOperation({ summary: "健康检查" })
  async health() {
    return ResponseUtil.success({
      status: "ok",
      timestamp: new Date().toISOString(),
      service: "乌东文旅 API",
      version: "1.0.0",
    });
  }
}

File: server-java/src/main/java/com/wudong/controller/HealthController.java
package com.wudong.controller;

import com.wudong.common.response.ApiResponse;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.time.LocalDateTime;
import java.util.HashMap;
import java.util.Map;

@Tag(name = "健康检查")
@RestController
public class HealthController {

    @Operation(summary = "健康检查")
    @GetMapping("/api/health")
    public ApiResponse<Map<String, Object>> health() {
        Map<String, Object> data = new HashMap<>();
        data.put("status", "ok");
        data.put("timestamp", LocalDateTime.now());
        data.put("version", "1.0.0");
        data.put("runtime", "Java + Spring Boot");
        return ApiResponse.success(data);
    }
}
